app/models/order_detail.rb

```diff
@@ -890,6 +890,15 @@ def notify_purchaser_of_order_status
     end
   end
 
+  def self.where_order_number_and_id(query)
+    return nil unless /\A(?<order_id>\d+)-(?<id>\d+)\z/ =~ query
+    where(id: id, order_id: order_id)
+  end
+
+  def to_log_s
+    "Order # #{order_id}-#{id}"
```

Comment:
Can we use `OrderDetail#order_number`? And this is kind of picky, but everywhere we display an order number there's no space between the `#` and the number.

Reply:
Sure